src/main/frontend/pipeline-console-view/pipeline-console/main/PipelineConsole.tsx

```diff
@@ -186,10 +190,14 @@ export class PipelineConsole extends React.Component<
       .then((step_res) => step_res.json())
       .then((step_result) => {
         console.debug("Updating steps");
-        console.debug(JSON.stringify(step_result.data));
-        this.setState({
-          steps: step_result.data.steps,
-        });
+        this.setState(
+          {
+            steps: step_result.data.steps,
+          },
+          () => {
+            this.handleUrlParams();
```

Comment:
Using the optional callback argument to `setState` allows `handleUrlParams` to be called after the state is actually updated. Unfortunately `handleUrlParams` is now called twice because we can't rely on the order of `setStages` or `setSteps` completing. There is likely a better way to wait on both calls to `setState` completing.